packages/plugin-conversation/src/decrypter.js

```diff
@@ -28,24 +28,23 @@ const Decrypter = SparkPlugin.extend({
       return Promise.resolve();
     }
 
-    if (isArray(object)) {
-      return Promise.all(object.map((item) => this.decryptObject(key, item)))
-        .then(() => object);
-    }
-
-    if (object.encryptionKeyUrl) {
-      key = object.encryptionKeyUrl;
-    }
-
-    if (object.objectType) {
-      const methodName = S(`decrypt_${object.objectType}`).camelize().s;
-      if (isFunction(this[methodName])) {
-        return this[methodName](key, object)
-          .then(() => object);
-      }
-    }
+    return this.handleArrayObject(key, object)
```

Comment:
If you broke this up for code complexity reasons, it may make more sense to keep the array handling inside decryptObject and move the object handling to its own function. in its current form, the object handling code will get called directly on object, even if object is an array.